internal/insights/insights.go

```
@@ -115,6 +137,84 @@ func GetLangStatsInsights(ctx context.Context, db dbutil.DB, filter SettingFilte
 	return results, nil
 }
 
+// GetIntegratedInsights returns all of the insights defined by the extension based Code Insights that are compatible
+// running over all repositories. These are located in a specific setting object `insights.allrepos` which is a
+// dictionary of unique keys to extension setting body. This is intended to be deprecated as soon as code insights migrates
+// fully to a persistent database. Any deserialization errors that occur during parsing will be logged as errors, but will not
+// cause any errors to surface.
+func GetIntegratedInsights(ctx context.Context, db dbutil.DB) ([]SearchInsight, error) {
+	prefix := "insights.allrepos"
+
+	settings, err := GetSettings(ctx, db, All, prefix)
+	if err != nil {
+		return []SearchInsight{}, err
+	}
+
+	var multi error
+
+	results := make([]SearchInsight, 0)
+	for _, setting := range settings {
+		var raw map[string]json.RawMessage
+		raw, err = FilterSettingJson(setting.Contents, prefix)
+		if err != nil {
+			multi = multierror.Append(multi, err)
+			continue
+		}
+
+		for _, val := range raw {
+			// iterate for each instance of the prefix key in the settings. This should never be len > 1, but it's technically a map.
+			temp, err := unmarshalIntegrated(val)
+			if err != nil {
+				// this isn't actually a total failure case, we could have partially parsed this dictionary.
+				multi = multierror.Append(multi, err)
+			}
+			results = append(results, temp.Insights()...)
+		}
+	}
+
+	if multi != nil {
+		log15.Error("deserialization errors parsing integrated insights", "error", multi)
```

```suggestion
		log15.Error("insights: deserialization errors parsing integrated insights", "error", multi)
```

---

maybe you want to return the error here as well?

---

In this case, we are silencing the error completely. It's kind of a weird thing, but there is no guarantee these settings are well-formed. So, we will throw out any that aren't well-formed. This is a temporary solution anyway, until we migrate all of these to a database.